### ISSUE TITLE
cached resolution: updateClassifiers does not download sources

### ISSUE THREAD

**Issue by OlegYch** [sbt/sbt#1776]
## steps

eg with this config and sbt 0.13.7

```
scalaVersion := "2.11.4"
//fork := true
libraryDependencies += "org.scalaz.stream" %% "scalaz-stream" % "snapshot-0.6a"
resolvers += "Scalaz Bintray Repo" at "http://dl.bintray.com/scalaz/releases"
transitiveClassifiers in updateClassifiers := List("sources")
updateOptions := updateOptions.value.withCachedResolution(true).withLatestSnapshots(false)
```
## problem

updateClassifiers will only download "bundles"
## expectation

without cached resolution it will also download "jars" and "srcs" (not sure why it downloads both bundles and jars, though)

**Comment by fommil** [comment]
:+1: this looks like it stings us in ensime-server

**Comment by nightscape** [comment]
This sounds like it could also be the problem for this sbteclipse issue:
https://github.com/typesafehub/sbteclipse/issues/234

**Comment by fommil** [comment]
FWIW, reverting to 0.13.7-RC3 works OK for us.

**Comment by mauhiz** [comment]
This issue also impacts IntelliJ's SBT plugin.

**Comment by fommil** [comment]
Funny you say that. I helped a colleague with this today using the gen-idea plugin. Workaround was to fall back to rc3

**Comment by dpratt** [comment]
I'm seeing this as well - I'd vote for this to get fixed in 0.13.8, if possible. It's pretty annoying.

**Comment by eed3si9n** [comment]
It seems to be fixed with 0.13.8-M6:

```
sbt-1776> updateClassifiers
[info] Updating {file:/Users/eugene/work/quick-test/sbt-1776/}sbt-1776...
[info] Done updating.
[warn] Multiple dependencies with the same organization/name but different versions. To avoid conflict, pick one version:
[warn]  * org.scala-lang:scala-reflect:(2.11.0, 2.11.4)
[info] Resolving org.scala-lang#scala-library;2.11.4 ...
[info] downloading https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.11.4/scala-library-2.11.4-sources.jar ...
[info]  [SUCCESSFUL ] org.scala-lang#scala-library;2.11.4!scala-library.jar(src) (1401ms)
[info] Resolving org.scalaz#scalaz-concurrent_2.11;7.1.0 ...
[info] downloading https://repo1.maven.org/maven2/org/scalaz/scalaz-concurrent_2.11/7.1.0/scalaz-concurrent_2.11-7.1.0-sources.jar ...
[info]  [SUCCESSFUL ] org.scalaz#scalaz-concurrent_2.11;7.1.0!scalaz-concurrent_2.11.jar(src) (272ms)
[info] downloading https://repo1.maven.org/maven2/org/scalaz/scalaz-concurrent_2.11/7.1.0/scalaz-concurrent_2.11-7.1.0.jar ...
[info]  [SUCCESSFUL ] org.scalaz#scalaz-concurrent_2.11;7.1.0!scalaz-concurrent_2.11.jar (316ms)
[info] Resolving org.scalaz#scalaz-core_2.11;7.1.0 ...
[info] downloading https://repo1.maven.org/maven2/org/scalaz/scalaz-core_2.11/7.1.0/scalaz-core_2.11-7.1.0.jar ...
[info]  [SUCCESSFUL ] org.scalaz#scalaz-core_2.11;7.1.0!scalaz-core_2.11.jar (1601ms)
[info] downloading https://repo1.maven.org/maven2/org/scalaz/scalaz-core_2.11/7.1.0/scalaz-core_2.11-7.1.0-sources.jar ...
[info]  [SUCCESSFUL ] org.scalaz#scalaz-core_2.11;7.1.0!scalaz-core_2.11.jar(src) (246ms)
[info] Resolving org.scala-lang#scala-reflect;2.11.0 ...
[info] downloading https://repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.11.0/scala-reflect-2.11.0-sources.jar ...
[info]  [SUCCESSFUL ] org.scala-lang#scala-reflect;2.11.0!scala-reflect.jar(src) (295ms)
[info] Resolving org.scalaz#scalaz-effect_2.11;7.1.0 ...
[info] downloading https://repo1.maven.org/maven2/org/scalaz/scalaz-effect_2.11/7.1.0/scalaz-effect_2.11-7.1.0.jar ...
[info]  [SUCCESSFUL ] org.scalaz#scalaz-effect_2.11;7.1.0!scalaz-effect_2.11.jar (326ms)
[info] downloading https://repo1.maven.org/maven2/org/scalaz/scalaz-effect_2.11/7.1.0/scalaz-effect_2.11-7.1.0-sources.jar ...
[info]  [SUCCESSFUL ] org.scalaz#scalaz-effect_2.11;7.1.0!scalaz-effect_2.11.jar(src) (252ms)
[info] Resolving org.typelevel#scodec-bits_2.11;1.0.4 ...
[info] downloading https://repo1.maven.org/maven2/org/typelevel/scodec-bits_2.11/1.0.4/scodec-bits_2.11-1.0.4.jar ...
[info]  [SUCCESSFUL ] org.typelevel#scodec-bits_2.11;1.0.4!scodec-bits_2.11.jar (227ms)
[info] downloading https://repo1.maven.org/maven2/org/typelevel/scodec-bits_2.11/1.0.4/scodec-bits_2.11-1.0.4-sources.jar ...
[info]  [SUCCESSFUL ] org.typelevel#scodec-bits_2.11;1.0.4!scodec-bits_2.11.jar(src) (294ms)
[info] Resolving org.scalaz.stream#scalaz-stream_2.11;snapshot-0.6a ...
[info] downloading http://dl.bintray.com/scalaz/releases/org/scalaz/stream/scalaz-stream_2.11/snapshot-0.6a/scalaz-stream_2.11-snapshot-0.6a-sources.jar ...
[info]  [SUCCESSFUL ] org.scalaz.stream#scalaz-stream_2.11;snapshot-0.6a!scalaz-stream_2.11.jar(src) (854ms)
[info] downloading http://dl.bintray.com/scalaz/releases/org/scalaz/stream/scalaz-stream_2.11/snapshot-0.6a/scalaz-stream_2.11-snapshot-0.6a.jar ...
[info]  [SUCCESSFUL ] org.scalaz.stream#scalaz-stream_2.11;snapshot-0.6a!scalaz-stream_2.11.jar (4211ms)
[info] Resolving org.scala-lang.modules#scala-parser-combinators_2.11;1.0.2 ...
[info] downloading https://repo1.maven.org/maven2/org/scala-lang/modules/scala-parser-combinators_2.11/1.0.2/scala-parser-combinators_2.11-1.0.2.jar ...
[info]  [SUCCESSFUL ] org.scala-lang.modules#scala-parser-combinators_2.11;1.0.2!scala-parser-combinators_2.11.jar (536ms)
[info] downloading https://repo1.maven.org/maven2/org/scala-lang/modules/scala-parser-combinators_2.11/1.0.2/scala-parser-combinators_2.11-1.0.2-sources.jar ...
[info]  [SUCCESSFUL ] org.scala-lang.modules#scala-parser-combinators_2.11;1.0.2!scala-parser-combinators_2.11.jar(src) (215ms)
[info] Resolving org.scala-lang.modules#scala-xml_2.11;1.0.2 ...
[info] downloading https://repo1.maven.org/maven2/org/scala-lang/modules/scala-xml_2.11/1.0.2/scala-xml_2.11-1.0.2-sources.jar ...
[info]  [SUCCESSFUL ] org.scala-lang.modules#scala-xml_2.11;1.0.2!scala-xml_2.11.jar(src) (257ms)
[info] downloading https://repo1.maven.org/maven2/org/scala-lang/modules/scala-xml_2.11/1.0.2/scala-xml_2.11-1.0.2.jar ...
[info]  [SUCCESSFUL ] org.scala-lang.modules#scala-xml_2.11;1.0.2!scala-xml_2.11.jar (353ms)
[info] Resolving org.scala-lang#scala-compiler;2.11.4 ...
[info] downloading https://repo1.maven.org/maven2/org/scala-lang/scala-compiler/2.11.4/scala-compiler-2.11.4-sources.jar ...
[info]  [SUCCESSFUL ] org.scala-lang#scala-compiler;2.11.4!scala-compiler.jar(src) (629ms)
[info] Resolving org.scala-lang#scala-reflect;2.11.4 ...
[info] downloading https://repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.11.4/scala-reflect-2.11.4-sources.jar ...
[info]  [SUCCESSFUL ] org.scala-lang#scala-reflect;2.11.4!scala-reflect.jar(src) (305ms)
[info] Resolving jline#jline;2.12 ...
[info] downloading https://repo1.maven.org/maven2/jline/jline/2.12/jline-2.12-sources.jar ...
[info]  [SUCCESSFUL ] jline#jline;2.12!jline.jar(src) (225ms)
[success] Total time: 14 s, completed Feb 25, 2015 11:34:57 PM
```

**Comment by fommil** [comment]
confirmed fixed in 0.13.8
